src/main/java/ch/njol/skript/conditions/CondHasItemCooldown.java

```diff
@@ -48,9 +53,18 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 	@Override
 	public boolean check(Event event) {
 		return players.check(event, (player) ->
-				itemtypes.check(event, (itemType) ->
-					itemType.hasType() && player.hasCooldown(itemType.getMaterial())
-				)
+				itemtypes.check(event, (itemType) -> {
```

Comment:
This needs to be evaluated outside the players check and use SimpleExpression.check() instead, so as to avoid evaluating itemtypes multiple times.